codebuild/bin/install_s2n_head.sh

```diff
@@ -12,31 +12,47 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-set -ex
-pushd "$(pwd)"
+set -e
 
 usage() {
     echo "install_s2n_head.sh build_dir"
     exit 1
 }
 
+BUILD_DIR=$1
+SRC_ROOT=${SRC_ROOT:-$(pwd)}
+
 if [ "$#" -ne "1" ]; then
     usage
 fi
 
-BUILD_DIR=$1
-source codebuild/bin/jobs.sh
-cd "$BUILD_DIR"
-
-# Clone the most recent s2n commit
-git clone --depth=1 https://github.com/aws/s2n-tls s2n_head
-cmake ./s2n_head -Bbuild -DCMAKE_PREFIX_PATH="$LIBCRYPTO_ROOT" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=on
-cmake --build ./build -- -j $JOBS
+# CMake(nix) and Make are using different directory structures.
+if [[ "$IN_NIX_SHELL" ]]; then
```

Comment:
I think I'm regretting my comment about combining these two scripts. It kind of seems like they have nothing in common after all. You keep branching on $IN_NIX_SHELL, and the s2nc_head existence check wouldn't be required in the CI either. Seems like they have zero code in common.